go-selinux/selinux_linux_test.go

```diff
@@ -412,6 +413,76 @@ func TestGlbLub(t *testing.T) {
 	}
 }
 
+func TestContextWithLevel(t *testing.T) {
+	want := "bob:sysadm_r:sysadm_t:SystemLow-SystemHigh"
+
+	verifier := func(con string) error {
+		if con != want {
+			return fmt.Errorf("invalid context %s", con)
+		}
+
+		return nil
+	}
+
+	userBuff := `# COMMENT
+foo_r:foo_t:s0     sysadm_r:sysadm_t:s0
+staff_r:staff_t:s0                 baz_r:baz_t:s0   sysadm_r:sysadm_t:s0
+`
+
+	defaultBuff := `
+foo_r:foo_t:s0     sysadm_r:sysadm_t:s0
+staff_r:staff_t:s0                 baz_r:baz_t:s0   sysadm_r:sysadm_t:s0
+`
+
+	c := defaultSECtx{
+		user:       "bob",
+		level:      "SystemLow-SystemHigh",
+		scon:       "system_u:staff_r:staff_t:s0",
+		userRdr:    bytes.NewBufferString(userBuff),
+		defaultRdr: bytes.NewBufferString(defaultBuff),
+		verifier:   verifier,
+	}
+
+	t.Run("match exists in user context file", func(t *testing.T) {
+		got, err := getDefaultContextFromReaders(c)
+		if err != nil {
+			t.Fatalf("err should not exist but is: %v", err)
+		}
+
+		if got != want {
+			t.Fatalf("got context: %q but expected %q", got, want)
+		}
+	})
+
+	t.Run("match exists in default context file, but not in user file", func(t *testing.T) {
+		userBuff = `# COMMENT
```

Comment:
You rewrite the `userBuff` here, so all the test cases below will use it. This smells buggy to me.